### PR TITLE
Added UNION clause

### PR DESCRIPTION
**Pull request by transistive**
UNION clauses are a bit special as they essentially combine two queries logically, yet the clauses are not structured hierarchically but sequentially.

Because of this, the UNION clause is just a clause with the subject possibly being `ALL`.

There are helper methods such as `UnionClause::union` and `Query::union` which accepts decorators and or query objects and then merge the clauses appropriately.

### CODE REVIEW

File: src/Clauses/UnionClause.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    /**
+     * Combines two queries with a union.
+     *
+     * @param Query $left The query preceding the union clause.
+     * @param Query $right The query after the union clause.
+     * @param bool $all Whether the union should include all results or remove the duplicates instead.
+     */
+    public static function union(Query $left, Query $right, bool $all = false): Query
+    {
+        $tbr = Query::new();
```
Comment (wgevaert) [inline src/Clauses/UnionClause.php:34]: Is there a reason why you do not just do `$tbr = clone $left` to start with? Readability?

**Review by wgevaert — APPROVED** [review]
Looks good!

File: src/Clauses/UnionClause.php (hunk 1 above)
Comment (transistive) [reply src/Clauses/UnionClause.php:34]: To me it's a lot more readable, never thought about cloning. If you want, I can change it?

Comment (wgevaert) [reply src/Clauses/UnionClause.php:34]: Nah I think it is okay this way, just wondering

**Review by wgevaert — APPROVED** [review]

**Comment by wgevaert** [comment]
Released in 4.2.0

**Pull request merged**